app/assets/javascripts/tasks_checklist.coffee

```diff
@@ -0,0 +1,141 @@
+MIN_TITLE_CHARS = 1
+MIN_DESCRIPTION_WORDS = 30
+MIN_LABELS = 2
+
+todos = ['title', 'programming_language', 'description', 'language', 'labels', 'license', 'file', 'test', 'model_solution']
+
+initialize_checklist = ->
+  if $('.completeness-checklist-container').length == 0
+    return
+
+  initialize_callbacks()
+  set_progress(0)
+  update_checklist()
+
+  $('.maximized-checklist').on 'click', -> minimize_checklist()
+  $('.minimized-checklist').on 'click', -> maximize_checklist()
+
+
+maximize_checklist = ->
+  if $('.maximized-checklist').hasClass('d-none')
+    toggle_checklist(200, 400)
+
+minimize_checklist = ->
+  if $('.minimized-checklist').hasClass('d-none')
+    toggle_checklist(400, 200)
+
+toggle_checklist = (closing_duration, opening_duration) ->
+  $('.checklist-content').animate({width: [ "toggle", "swing" ]}, closing_duration, () ->
+    $(this).find('.maximized-checklist, .minimized-checklist').toggleClass('d-none');
+    $('.checklist-content').animate({width: [ "toggle", "swing" ]}, opening_duration)
+  )
+
+
+click_handler = ->
+  setTimeout -> # trick to execute after all other handlers (after new nested file has been created)
+    update_checklist()
+    $('.btn.remove_nested_fields_link').on 'click', (_) -> setTimeout update_checklist, 0
+    $('.add_nested_fields_link').off 'click', click_handler
+    $('.add_nested_fields_link').on 'click', click_handler
+  , 0
+
+initialize_callbacks = ->
+  $('body').on 'change', (_) -> update_checklist()
+  $('.add_nested_fields_link').on 'click', click_handler
+  $('.btn.remove_nested_fields_link').on 'click', (_) -> setTimeout update_checklist, 0
+
+
+check = (todo_name) ->
+  switch
+    when todo_name == 'title' then $('#task_title').val().length >= MIN_TITLE_CHARS
+    when todo_name == 'programming_language' then $('#task_programming_language_id').val().length > 0
+    when todo_name == 'description' then $('#markdown-input-description').val().trim().split(/\s+/).length > MIN_DESCRIPTION_WORDS
+    when todo_name == 'language' then $('#task_language').val().length > 0
+    when todo_name == 'labels' then $('#task_label_names').val().length >= MIN_LABELS
+    when todo_name == 'license' then $('#task_license_id').val().length > 0
+    when todo_name == 'file' then check_files($('fieldset.nested_task_files'))
+    when todo_name == 'test' then check_tests()
+    when todo_name == 'model_solution' then check_model_solutions()
+    else console.error("Unknown todo_name #{todo_name}")
+
+
+check_files = (files) ->
+  ok = false
+
+  files.each (index, file) ->
+    if $(file).css('display') != 'none'
+
+      name_present = $(file).find('.file-name').val().length > 0
+      content_present = $(file).find('.file-content').val().length > 0
```

Comment:
At least for me (in Safari and Chrome), the `content_present` check is not working reliably (and an empty content is accepted).

Reply:
Actually `attachment_present` was the problem. I fixed the CSS selector. It works for me, can you test it on your machine again?